feature-libs/checkout/base/components/checkout-delivery-mode/checkout-delivery-mode.component.ts

```diff
@@ -100,6 +101,32 @@ export class CheckoutDeliveryModeComponent {
     protected activeCartFacade: ActiveCartFacade
   ) {}
 
+  /**
+   * We capture the keyboard activated option,
+   * then restore the focus to it after the data is updated.
+   */
+  updateMethod(code: string | undefined, event: Event): void {
```

Comment:
Is it possible to move this logic into the `changeMode` method? We can add a protected method or something to handle the refocusing. Otherwise it looks like we are in-reality deprecating the `changeMode` method which doesn't really make sense (`updateMethod` still does all `changeMode` does but "fixed"). 

In doing this, we should include the `$event` parameter in the `changeMode` method as an optional parameter and we can extend its logic if it is passed.

Reply:
I've refactored it, should be easier to read now.